frontend/src/app/sessions/user-sessions-wrapper/create-session/create-readonly-session/create-readonly-session.component.ts

```diff
@@ -99,14 +150,15 @@ export class CreateReadonlySessionComponent implements OnInit {
   onToolChange(tool: Tool): void {
     this.toolSelectionForm.controls.version.enable();
     this.toolSelectionForm.controls.version.patchValue(null);
-    this.toolVersions = undefined;
 
-    this.toolService.getVersionsForTool(tool.id).subscribe((toolVersions) => {
-      const toolVersionIds = this.models?.map((m) => m.version?.id);
-      this.toolVersions = toolVersions.filter((v) =>
-        toolVersionIds?.includes(v.id),
+    this.relevantToolVersions = this.models
+      ?.map((m) => m.compatibleVersions)
+      .flat()
+      .filter((v) => v.tool.id === tool.id)
+      .filter(
+        (value, index, self) =>
+          index === self.findIndex((version) => version.id === value.id),
```

Comment:
Similar to above, please either use a proper function here, make this more understandable, or add a comment explaining what and why this is done here.